### PR TITLE
Implement CloseTrades_cBot

### PR DESCRIPTION
**Pull request by LucyQuantAnalyst**

### CODE REVIEW

File: cBots/CloseTrades_cBot.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+               Print("Finished closing orders and positions");
+            }
+            
+            Print("Finished onBar step"); 
+        }
+        
+        private void CancelOrdersAndPositionsForAllMarkets()
+        {
+            Print("Cancel all orders and positions for all markets");
+            var pendingOrders = PendingOrders;
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:82]: Hotovo

File: cBots/CloseTrades_cBot.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            foreach (var order in pendingOrders)
+            {
+                CancelPendingOrder(order);
+            }
+
+            var openPositions = Positions;
+            foreach (var position in openPositions)
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:89]: Hotovo

File: cBots/CloseTrades_cBot.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                    }
+               }
+               
+               Print("Finished closing orders and positions");
+            }
+            
+            Print("Finished onBar step"); 
+        }
+        
+        private void CancelOrdersAndPositionsForAllMarkets()
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:79]: Hotovo

File: cBots/CloseTrades_cBot.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            }
+
+            var openPositions = Positions;
+            foreach (var position in openPositions)
+            {
+                 ClosePosition(position);
+            }
+        }
+        
+        private void CancelOrdersAndPositions(string market)
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:95]: Hotovo

File: cBots/CloseTrades_cBot.cs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        [Parameter(DefaultValue = 0)]
+        public int Minutes {get; set;}
+
+        [Parameter(DefaultValue = true)]
+        public bool All {get; set;}
+
+        [Parameter(DefaultValue = "Copper")]
+        public string Market {get; set;}
+
+        private TimeOnly CloseTime {get; set;}
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:38]: Hotovo

File: cBots/CloseTrades_cBot.cs [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            Print("Start CloseTrades_cBot");
+        }
+
+        protected override void OnBar()
+        {
+            Print("Start onBar step");
+
+            CloseTime = new TimeOnly(Hours, Minutes);
+             
+            Print("Close time: {0}", CloseTime);
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:51]: Hotovo

File: cBots/CloseTrades_cBot.cs [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+Date: 29.10.2023
+Version: 1.0.3
+*/
+
+namespace cAlgo.Robots
+{
+    [Robot(AccessRights = AccessRights.None)]
+    public class CloseTrades_cBot : Robot
+    {   
+        [Parameter(DefaultValue = DayOfWeek.Friday)]
```
Comment (GeorgeQuantAnalyst) [inline cBots/CloseTrades_cBot.cs:23]: Pridat komentar

// User defined properties

File: cBots/CloseTrades_cBot.cs [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        [Parameter(DefaultValue = 0)]
+        public int Minutes {get; set;}
+
+        [Parameter(DefaultValue = true)]
+        public bool All {get; set;}
+
+        [Parameter(DefaultValue = "Copper")]
+        public string Market {get; set;}
+
+        private TimeOnly CloseTime {get; set;}
```
Comment (GeorgeQuantAnalyst) [inline cBots/CloseTrades_cBot.cs:38]: //Constants

File: cBots/CloseTrades_cBot.cs [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+            List<String> inputErrorMessages = ValidateInputs();
+            inputErrorMessages.ForEach(m => Print(m));
+            if (inputErrorMessages.Any()){
+                Print("App contains input validation errors and will be stop.");
+                Stop();
+                return;
+            }
+            Print("Finished validation of User defined properties ...");
+            
+            CloseTime = new TimeOnly(Hours, Minutes);
```
Comment (GeorgeQuantAnalyst) [inline cBots/CloseTrades_cBot.cs:61]: Print("Compute properties ...");

...

Print("Computed properties:");

...

File: cBots/CloseTrades_cBot.cs (hunk 7 above)
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:23]: Hotovo

File: cBots/CloseTrades_cBot.cs (hunk 9 above)
Comment (LucyQuantAnalyst) [reply cBots/CloseTrades_cBot.cs:61]: Hotovo

**Pull request merged**